packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.swift

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import Foundation
+
+public final class IosPlatformImagesPlugin: NSObject, FlutterPlugin {
+
+  private let channel: MethodChannel
+
+  init(channel: MethodChannel) {
+    self.channel = channel
+  }
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/ios_platform_images",
+      binaryMessenger: registrar.messenger())
+
+    let instance = IosPlatformImagesPlugin(channel: channel)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    switch call.method {
+    case "loadImage":
+      loadImage(call, result)
+    case "resolveURL":
+      resolveURL(call, result)
+    default:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+  private func loadImage(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
+    guard let arguments = call.arguments as? [Any],
+      let name = arguments.first as? String,
+      let image = UIImage.flutterImage(withName: name)
+    else {
+      result(nil)
+      return
+    }
+
+    let data = image.pngData()
+
+    let imageResult: [String: Any] = [
+      "scale": image.scale,
+      "data": FlutterStandardTypedData(bytes: data!),
+    ]
+
+    result(imageResult)
+  }
+
+  private func resolveURL(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
+    if let args = call.arguments as? [String?] {
+      let name = args[0]
+      let extensionName = args.count == 2 ? args[1] : nil
```

Comment:
This is not the same logic as the Obj-C version, and is not correct. There will always be two arguments, but the second may be `NSNull`.

Reply:
I change to avoid  `Fatal error: Index out of range`

Reply:
In what context? [This](https://github.com/flutter/packages/blob/59d7b6d86c77ac971dd14b8c92bad75837515b22/packages/ios_platform_images/lib/ios_platform_images.dart#L147) is the call, and it passes two arguments.

If you are referring to the unit tests you added, the solution is to make the unit tests match actual usage. Having unit tests that call the method in a way that production code never will, and then changing the implementation to work in the unit test but break production, makes the unit tests actively harmful.

Reply:
I saw here in another test the reason for your comment, I will make the appropriate change